File: dotnet/src/MyDotey.SCF.Simple/Type/String/StringToFloatConverter.cs
using System;

namespace MyDotey.SCF.Type.String
{
    /**
     * @author koqizhao
     *
     * May 21, 2018
     */
    public class StringToFloatConverter : StringConverter<float?>
    {
        public static readonly StringToFloatConverter DEFAULT = new StringToFloatConverter();

        public override float? Convert(string source)
        {
            if (string.IsNullOrWhiteSpace(source))
                return null;

            return float.Parse(source);
        }
    }
}

File: java/scf-simple/src/main/java/org/mydotey/scf/type/string/StringToFloatConverter.java
package org.mydotey.scf.type.string;

/**
 * @author koqizhao
 *
 * May 21, 2018
 */
public class StringToFloatConverter extends StringConverter<Float> {

    public static final StringToFloatConverter DEFAULT = new StringToFloatConverter();

    public StringToFloatConverter() {
        super(Float.class);
    }

    @Override
    public Float convert(String source) {
        return Float.parseFloat(source);
    }

}
